frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -1,18 +1,202 @@
 import { MemberType } from "../../hooks/UseOktaMemberships";
+import { UseOrganizationSenders } from "../../hooks/UseOrganizationSenders";
+import { RSSender } from "../../config/endpoints/settings";
+import useManagePublicKey, {
+    ManagePublicKeyActionType,
+} from "../../hooks/network/ManagePublicKey/ManagePublicKeyHooks";
+import { ContentType } from "../../utils/TemporarySettingsAPITypes";
+
+import { ManagePublicKeyUpload } from "./ManagePublicKeyUpload";
+
+const LightbulbIcon = Icon.Lightbulb;
+
+// TODO: create common component???
+const ManagePublicKeySpinner = ({ message }: { message: ReactNode }) => (
+    <div className="grid-col flex-1 display-flex flex-column flex-align-center margin-top-10">
+        <div className="grid-row">
+            <Spinner />
+        </div>
+        <div className="text-center">{message}</div>
+    </div>
+);
+
+const SendersDisplay = ({ senders }: { senders: RSSender[] }) => {
+    const navigate = useNavigate();
+    const { state, dispatch } = useManagePublicKey();
+    const [fileContent, setFileContent] = useState("");
+    const [selectedSender, setSelectedSender] = useState(
+        senders.length === 1 ? senders[0].name : ""
+    );
+    const [showUploadDisplay, setShowUploadDisplay] = useState(
+        senders.length === 1 ? true : false
+    );
+    const { fileInputResetValue, fileName, localError } = state;
+
+    const handleSenderSubmit = () => {
+        setShowUploadDisplay(true);
+    };
+
+    // TODO: mocked for now - make the call you need when sending the file
+    const { sendFile, isWorking } = {
+        // eslint-disable-next-line no-empty-pattern
+        sendFile: ({}) => {
+            return { fileName: "fileName" };
+        },
+        isWorking: false,
+    };
+
+    const resetState = () => {
+        setFileContent("");
+        dispatch({ type: ManagePublicKeyActionType.RESET });
+    };
+
+    const handleFileChange = async (
+        event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+        // No file selected
+        if (!event?.target?.files?.length) return;
+
+        const file = event.target.files.item(0);
+        if (!file) return;
+
+        const content = await file.text();
+
+        setFileContent(content);
+
+        dispatch({
+            type: ManagePublicKeyActionType.FILE_SELECTED,
+            payload: { file },
+        });
+    };
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        if (fileContent.length === 0) {
+            showError("No file contents to validate.");
+            return;
+        }
+
+        // initializes necessary state and sets `isSubmitting`
+        dispatch({ type: ManagePublicKeyActionType.PREPARE_FOR_REQUEST });
+
+        try {
+            sendFile({
+                contentType: ContentType.PEM,
+                fileContent: fileContent,
+                fileName: fileName,
+                selectedSender: selectedSender,
```

Comment:
Just learned that we will not need to send the selectedSender to the POST for this iteration.